service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Query.java

```diff
@@ -142,23 +144,42 @@ private void continueQuery(String continuationToken, int pageSize) throws IOExce
         this.pageSize = pageSize;
         this.requestContinuationToken = continuationToken;
         //Codes_SRS_QUERY_25_018: [The method shall send the query request again.]
-        sendQueryRequest(this.iotHubConnectionString, this.url, this.httpMethod, this.timeout);
+        sendQueryRequest(this.iotHubConnectionString, this.url, this.httpMethod, this.connectTimeoutInMs, this.responseTimeoutInMs);
+    }
+
+    /**
+     * Sends request for the query to the IotHub with default timeout
+     * @param iotHubConnectionString Hub Connection String
+     * @param url URL to Query on
+     * @param method HTTP Method for the requesting a query
+     * @return QueryResponse object which holds the response Iterator
+     * @throws IOException If any of the input parameters are not valid
+     * @throws IotHubException If HTTP response other then status ok is received
+     */
+    public QueryResponse sendQueryRequest(IotHubConnectionString iotHubConnectionString,
+                                          URL url,
+                                          HttpMethod method) throws IOException, IotHubException
+    {
+      return sendQueryRequest(iotHubConnectionString, url, method, DeviceOperations.DEFAULT_CONNECT_TIMEOUT_MS, DeviceOperations.DEFAULT_RESPONSE_TIMEOUT_MS);
     }
 
     /**
      * Sends request for the query to the IotHub
      * @param iotHubConnectionString Hub Connection String
      * @param url URL to Query on
      * @param method HTTP Method for the requesting a query
-     * @param timeoutInMs Maximum time to wait for the hub to respond
+     * @param connectTimeoutInMs Maximum time to wait for the hub to be connected
+     * @param responseTimeoutInMs Maximum time to wait for the hub to respond
      * @return QueryResponse object which holds the response Iterator
      * @throws IOException If any of the input parameters are not valid
      * @throws IotHubException If HTTP response other then status ok is received
      */
     public QueryResponse sendQueryRequest(IotHubConnectionString iotHubConnectionString,
                                    URL url,
                                    HttpMethod method,
-                                   Long timeoutInMs) throws IOException, IotHubException
+                                   Long connectTimeoutInMs,
```

Comment:
This is a breaking change as you are modifying a public API. My suggestion is to leave the existing API as is and create a new one with additional parameters

Reply:
keep the existing API, but change the parameter name _timeoutInMs_ to _responseTimeoutInMs_